File: guides/java/kora-java-guide-database-cassandra-app/src/main/java/ru/tinkoff/kora/guide/databasecassandra/repository/UserRepository.java
package ru.tinkoff.kora.guide.databasecassandra.repository;

import jakarta.annotation.Nullable;
import java.util.List;
import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.cassandra.CassandraRepository;

@Repository
public interface UserRepository extends CassandraRepository {

    @Query("SELECT id, name, email, created_at FROM users")
    List<UserDAO> findAll();

    @Query("SELECT id, name, email, created_at FROM users WHERE id = :id")
    @Nullable
    UserDAO findById(String id);

    @Query("""
            INSERT INTO users(id, name, email, created_at)
            VALUES (:user.id, :user.name, :user.email, :user.createdAt)
            """)
    void save(UserDAO user);

    @Query("""
            UPDATE users
            SET name = :user.name, email = :user.email, created_at = :user.createdAt
            WHERE id = :user.id
            """)
    void update(UserDAO user);

    @Query("DELETE FROM users WHERE id = :id")
    void deleteById(String id);
}

File: guides/kotlin/kora-kotlin-guide-database-cassandra-app/src/main/kotlin/ru/tinkoff/kora/guide/databasecassandra/repository/UserRepository.kt
package ru.tinkoff.kora.guide.databasecassandra.repository

import ru.tinkoff.kora.database.cassandra.CassandraRepository
import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository

@Repository
interface UserRepository : CassandraRepository {

    @Query("SELECT id, name, email, created_at FROM users")
    fun findAll(): List<UserDAO>

    @Query("SELECT id, name, email, created_at FROM users WHERE id = :id")
    fun findById(id: String): UserDAO?

    @Query(
        """
        INSERT INTO users(id, name, email, created_at)
        VALUES (:user.id, :user.name, :user.email, :user.createdAt)
        """
    )
    fun save(user: UserDAO)

    @Query(
        """
        UPDATE users
        SET name = :user.name, email = :user.email, created_at = :user.createdAt
        WHERE id = :user.id
        """
    )
    fun update(user: UserDAO)

    @Query("DELETE FROM users WHERE id = :id")
    fun deleteById(id: String)
}
